### PR TITLE
Self contain enrollment component

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-2396](https://openedx.atlassian.net/browse/PROD-2396)
Self-containing enrollment component by moving the api call inside the component. 
The only visual change this PR adds is to add a loader while the component fetches the results.

Screenshot 2021-07-09 at 11 38 40 AM

### CODE REVIEW

File: src/users/enrollments/Enrollments.jsx [hunk 1]
```diff
@@ -222,6 +233,8 @@ export default function Enrollments({
           />
         ) : (<React.Fragment key="nothing" />) }
       </TransitionReplace>
+      {!enrollmentData && <PageLoading srMessage="Loading" />}
+      {enrollmentData && (
       <Collapsible title={`Enrollments (${tableData.length})`} defaultOpen={expanded}>
```
Comment (Ali-D-Akbar) [inline src/users/enrollments/Enrollments.jsx:236]: a tab needs to be added here

File: src/users/enrollments/Enrollments.test.jsx [hunk 2]
```diff
@@ -14,20 +17,31 @@ const EnrollmentPageWrapper = (props) => (
...
+  const testUsername = 'edX';
+  const enrollmentWrapperData = {
+    user: 'edX',
+    changeHandler: jest.fn(() => {}),
+    expanded: true,
+  };
+
+  beforeEach(async () => {
+    mockAdapter = new MockAdapter(getAuthenticatedHttpClient());
+    mockAdapter.onGet(getEnrollmentsUrl(testUsername)).reply(200, enrollmentsData);
```
Comment (Ali-D-Akbar) [inline src/users/enrollments/Enrollments.test.jsx]: Must be changed with `jest.spyOn(api, 'getEnrollments').mockImplementationOnce(() => Promise.resolve(enrollmentsData));`

File: src/users/enrollments/Enrollments.test.jsx [hunk 3]
```diff
@@ -15,19 +14,27 @@ const EnrollmentPageWrapper = (props) => (
 
 describe('Course Enrollments Listing', () => {
   let wrapper;
-
-  beforeEach(() => {
-    wrapper = mount(<EnrollmentPageWrapper {...enrollmentsData} />);
+  const enrollmentWrapperData = {
```
Comment (azanbinzahid) [inline src/users/enrollments/Enrollments.test.jsx]: nit: `props`

File: src/users/enrollments/Enrollments.jsx [hunk 4]
```diff
@@ -223,29 +234,31 @@ export default function Enrollments({
         ) : (<React.Fragment key="nothing" />) }
       </TransitionReplace>
       <Collapsible title={`Enrollments (${tableData.length})`} defaultOpen={expanded}>
-        <Table
-          className="w-auto"
-          data={tableDataSortable.sort(
-            (firstElement, secondElement) => sort(firstElement, secondElement, sortColumn, sortDirection),
+        {!enrollmentData
```
Comment (azanbinzahid) [inline src/users/enrollments/Enrollments.jsx]: nit: reverse the condition by removing `!`

**Review by azanbinzahid — APPROVED** [review]

File: src/users/enrollments/Enrollments.jsx [hunk 5]
```diff
@@ -39,48 +43,55 @@ export default function Enrollments({
     setEnrollmentExtraData(extraData);
   }
 
+  useEffect(() => {
+    getEnrollments(user).then((result) => {
+      const camelCaseResult = camelCaseObject(result);
+      setEnrollmentData(camelCaseResult);
+    });
+  }, [user]);
```
Comment (DawoudSheraz) [inline src/users/enrollments/Enrollments.jsx:51]: IF any error occurs during enrollment fetch, what will the user see?

Comment (Ali-D-Akbar) [reply src/users/enrollments/Enrollments.jsx:51]: This page as it can only occur in error 500. 
Screenshot 2021-07-09 at 2 28 15 PM

Comment (DawoudSheraz) [reply src/users/enrollments/Enrollments.jsx:51]: we would want some error handling with the alert listing component. Not required for this PR but create a task to add proper error handling. We can't allow the page to break if one component does not work.

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**